include/bl_storage.h

```diff
@@ -7,15 +7,64 @@
 #ifndef BL_STORAGE_H_
 #define BL_STORAGE_H_
 
-#include <zephyr/types.h>
 #include <string.h>
+#include <zephyr/types.h>
+#include <drivers/nrfx_common.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 
+#if defined(CONFIG_IS_SECURE_BOOTLOADER) || defined(CONFIG_SECURE_BOOT)
+#include <pm_config.h>
+#define PROVISION_PARTITION_ADDRESS PM_PROVISION_ADDRESS
+#else
+#define PROVISION_PARTITION_ADDRESS CONFIG_SECURE_BOOT_STORAGE_START_ADDRESS
```

Comment:
I don't get it. How can NSIB not already know where it's storing it's stuff.

Why do we need this Kconfig?